networkx/utils/backends.py

```diff
@@ -1481,7 +1504,7 @@ def _can_convert(self, backend_name, graph_backend_names):
     def _does_backend_have(self, backend_name):
         """Does the specified backend have this algorithm?"""
         if backend_name == "networkx":
-            return True
+            return "networkx" in self.backends
```

Comment:
@dschult, this is the key location to understand how this PR affects dispatch logic. `_does_backend_have` is the method that answers whether a backend _has_ a function. NetworkX "has" a function only if "networkx" is in `self.backends`.

Other key changes are basically updating `if nx.config.fallback_to_nx` to `if nx.config.fallback_to_nx and "networkx" in self.backends`, which we could write as `if nx.config.fallback_to_nx and self._does_backend_have("networkx")`.

Reply:
Thanks for the pointer to this function. At the moment, it doesn't seem so central to the PR though -- the function `_does_backend_have` is only used to set up the error messages for `NotImplemented` cases.

But the phrase `"networkx" in self.backends` is key -- as you explain -- because it combines with `fallback_to_nx` to determine whether the networkx function is called. But couldn't this just be done by calling the networkx function to get the NotImplemented rather than storing and looking up this info in `self.backends`?  I'm sure it is more complicated than that. There are always side-effects.

Reply:
> ... `_does_backend_have` is only used ...

Touche. Good catch. I also (just now) updated `_can_backend_run` to return False if `"networkx" not in self.backends"`.